tests/test_lib_policy.py

```diff
@@ -1577,34 +1576,52 @@ def test_41_list_policies_user_realm_resolver(self):
         set_policy(name="scopeA_r_realmA_userA_resolverB", scope="scopeA", action="read", realm="realmA",
                    user="userA", resolver="resolverB")
         P = PolicyClass()
+
         # get policies for action read
         policies = P.list_policies(action="read")
         self.assertEqual(10, len(policies))
+
         # get policies applicable for realm A
         policies = P.list_policies(action="read", realm="realmA")
+        policy_names = [p["name"] for p in policies]
+        policy_names.sort()
         self.assertEqual(7, len(policies))
         correct_policies = ["scopeA_r", "scopeA_r_realmA", "scopeA_r_realmA_userA", "scopeA_r_resolverA",
-                            "scopeA_r_resolverB", "scopeA_r_realmA_userA_resolverA", "scopeA_r_realmA_userA_resolverB"]
-        self.assertListEqual(correct_policies, [p["name"] for p in policies])
+                            "scopeA_r_resolverB", "scopeA_r_realmA_userA_resolverA",
+                            "scopeA_r_realmA_userA_resolverB"]
+        correct_policies.sort()
+        self.assertListEqual(correct_policies, policy_names)
```

Comment:
If the policy names are unique You can just use sets, should be easier to compare.